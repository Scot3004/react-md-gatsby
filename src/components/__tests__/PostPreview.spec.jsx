import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostPreview from '../PostPreview';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => Component => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: (text) => text };
    return Component;
  },
}));

const postInfo = {
  "cover": "/logos/logo-1024.png",
  "date": "18/07/2017",
  "excerpt": "This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to vâ€¦",
  "path": "/introducing-gatsby-material-starter",
  "tags": [
    "gatsby",
    "material",
    "starter",
  ],
  "timeToRead": 2,
  "title": "Introducing Gatsby Material Starter",
}

it('renders correctly', () => {
  const tree = shallow(<PostPreview postInfo={postInfo} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly Mobile', () => {
  const testsWidth = global.innerWidth;
  global.innerWidth = 600;
  const tree = shallow(<PostPreview postInfo={postInfo} />);
  expect(toJson(tree)).toMatchSnapshot();
  global.innerWidth = testsWidth;
});


it('renders correctly relative cover', () => {
  postInfo.cover = "postcover";
  const tree = shallow(<PostPreview postInfo={postInfo} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('componentWillUnmount should be called on unmount', () => {
  const component = shallow(<PostPreview postInfo={postInfo} />);
  const componentWillUnmount = jest.spyOn(component.instance(), 'componentWillUnmount');
  component.unmount();
  expect(componentWillUnmount).toHaveBeenCalled();
});
