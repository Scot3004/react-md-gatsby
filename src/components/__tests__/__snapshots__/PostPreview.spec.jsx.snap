// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Card
  animate={true}
  className="md-grid md-cell md-cell--12"
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  expanderTooltipPosition="left"
  key="/introducing-gatsby-material-starter"
  raise={true}
>
  <GatsbyLink
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/introducing-gatsby-material-starter"
  >
    <Media
      aspectRatio="16-9"
      className="post-preview-cover"
      component="section"
      forceAspect={true}
      style={
        Object {
          "backgroundImage": "url(http://localhost/logos/logo-1024.png)",
          "height": "225px",
        }
      }
    >
      <MediaOverlay
        component="div"
      >
        <CardTitle
          title="Introducing Gatsby Material Starter"
        >
          <withInk(withTooltip(Button))
            className="md-cell--right"
            inkTransitionEnterTimeout={450}
            inkTransitionLeaveTimeout={300}
            inkTransitionOverlap={150}
            raised={true}
            secondary={true}
          >
            read
          </withInk(withTooltip(Button))>
        </CardTitle>
      </MediaOverlay>
    </Media>
  </GatsbyLink>
  <CardTitle
    avatar={
      <Avatar
        icon={
          <FontIcon
            iconClassName="fa fa-calendar"
          />
        }
        suffixes={
          Array [
            "red",
            "pink",
            "purple",
            "deep-purple",
            "indigo",
            "blue",
            "light-blue",
            "cyan",
            "teal",
            "green",
            "light-green",
            "lime",
            "yellow",
            "amber",
            "orange",
            "deep-orange",
            "brown",
            "grey",
            "blue-grey",
          ]
        }
      />
    }
    expander={false}
    subtitle="readTime"
    title="published on"
  />
  <CardText
    component="section"
    expandable={false}
  >
    This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…
    <PostTags
      tags={
        Array [
          "gatsby",
          "material",
          "starter",
        ]
      }
    />
  </CardText>
</Card>
`;

exports[`renders correctly Mobile 1`] = `
<Card
  animate={true}
  className="md-grid md-cell md-cell--12"
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  expanderTooltipPosition="left"
  key="/introducing-gatsby-material-starter"
  raise={true}
>
  <GatsbyLink
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/introducing-gatsby-material-starter"
  >
    <Media
      aspectRatio="16-9"
      className="post-preview-cover"
      component="section"
      forceAspect={true}
      style={
        Object {
          "backgroundImage": "url(http://localhost/logos/logo-1024.png)",
          "height": "162px",
        }
      }
    >
      <MediaOverlay
        component="div"
      >
        <CardTitle
          title="Introducing Gatsby Material Starter"
        >
          <withInk(withTooltip(Button))
            className="md-cell--right"
            inkTransitionEnterTimeout={450}
            inkTransitionLeaveTimeout={300}
            inkTransitionOverlap={150}
            raised={true}
            secondary={true}
          >
            read
          </withInk(withTooltip(Button))>
        </CardTitle>
      </MediaOverlay>
    </Media>
  </GatsbyLink>
  <CardTitle
    avatar={
      <Avatar
        icon={
          <FontIcon
            iconClassName="fa fa-calendar"
          />
        }
        suffixes={
          Array [
            "red",
            "pink",
            "purple",
            "deep-purple",
            "indigo",
            "blue",
            "light-blue",
            "cyan",
            "teal",
            "green",
            "light-green",
            "lime",
            "yellow",
            "amber",
            "orange",
            "deep-orange",
            "brown",
            "grey",
            "blue-grey",
          ]
        }
      />
    }
    expander={true}
    subtitle="readTime"
    title="published on"
  />
  <CardText
    component="section"
    expandable={true}
  >
    This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…
    <PostTags
      tags={
        Array [
          "gatsby",
          "material",
          "starter",
        ]
      }
    />
  </CardText>
</Card>
`;

exports[`renders correctly relative cover 1`] = `
<Card
  animate={true}
  className="md-grid md-cell md-cell--12"
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  expanderTooltipPosition="left"
  key="/introducing-gatsby-material-starter"
  raise={true}
>
  <GatsbyLink
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/introducing-gatsby-material-starter"
  >
    <Media
      aspectRatio="16-9"
      className="post-preview-cover"
      component="section"
      forceAspect={true}
      style={
        Object {
          "backgroundImage": "url(postcover)",
          "height": "225px",
        }
      }
    >
      <MediaOverlay
        component="div"
      >
        <CardTitle
          title="Introducing Gatsby Material Starter"
        >
          <withInk(withTooltip(Button))
            className="md-cell--right"
            inkTransitionEnterTimeout={450}
            inkTransitionLeaveTimeout={300}
            inkTransitionOverlap={150}
            raised={true}
            secondary={true}
          >
            read
          </withInk(withTooltip(Button))>
        </CardTitle>
      </MediaOverlay>
    </Media>
  </GatsbyLink>
  <CardTitle
    avatar={
      <Avatar
        icon={
          <FontIcon
            iconClassName="fa fa-calendar"
          />
        }
        suffixes={
          Array [
            "red",
            "pink",
            "purple",
            "deep-purple",
            "indigo",
            "blue",
            "light-blue",
            "cyan",
            "teal",
            "green",
            "light-green",
            "lime",
            "yellow",
            "amber",
            "orange",
            "deep-orange",
            "brown",
            "grey",
            "blue-grey",
          ]
        }
      />
    }
    expander={false}
    subtitle="readTime"
    title="published on"
  />
  <CardText
    component="section"
    expandable={false}
  >
    This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…
    <PostTags
      tags={
        Array [
          "gatsby",
          "material",
          "starter",
        ]
      }
    />
  </CardText>
</Card>
`;
